### PR TITLE
Add pm-cpu to maint-1.0, including GNU and Intel compilers

### PR DESCRIPTION
**Pull request by ndkeen**
Add pm-cpu as a machine to maint-1.0.
Add GNU v11 and Intel compilers.
Includes notes to make manual copies for issues not yet resolved in submodules.

Fixes https://github.com/E3SM-Project/E3SM/issues/5280

### CODE REVIEW

File: notes.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+There are build errors with some cases using Intel that can be resolved using a more
+recent xml_stream_parser.c.  The one here in this directory is from a recent version
+of E3SM main.
+
+cp xml_stream_parser.c components/mpas-source/src/framework/xml_stream_parser.c
+
+The build error without this change might be:
+xml_stream_parser.c:1732:6: error: call to undeclared function 'stream_mgr_add_field_c'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
+                                        stream_mgr_add_field_c(manager, streamID, (const char *)fieldname, packages_local, &err);
+                                        ^
```
Comment (rljacob) [inline notes.txt:14]: @jonbob can you make the above changes to the MPAS submodule version pointed to by maint-1.0?  Just need intel working on pm-cpu.

**Comment by ndkeen** [comment]
I thought I did, but obviously did not. Pushed now.

**Comment by ndkeen** [comment]
Merge this now?

**Comment by rljacob** [comment]
Yes go ahead.

**Pull request merged**